src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -61,9 +61,31 @@ class MetadataService(
   // fetches entities
   def getEntities(path: NetworkPath): Option[List[Entity]] = entities.get(path)
 
+  // gets current entities
+  def getCurrentEntities(path: NetworkPath): Future[Option[List[Entity]]] = {
+    platformDiscoveryOperations.getEntities.map { allEntities =>
```

Comment:
I think we can remove a redundant future mapping if the path check is done right out
```scala
  if (exists(path))
    platformDiscoveryOperations.getEntities.map { allEntities =>
        Some(transformation.overrideEntities(path, allEntities, shouldLog = false))
    }
  else Future.successful(None)
```